source/patterns/@aws-solutions-constructs/aws-fargate-stepfunctions/test/fargate-stepfunctions.test.ts

```diff
@@ -0,0 +1,463 @@
+  return {
+    Family: familyName,
+    ContainerDefinitions: [
+      {
+        Environment: [
+          {
+            Name: "STATE_MACHINE_ARN",
+            Value: {
+              Ref: "testconstructStateMachine3333AAA9"
+            }
+          }
+        ],
+        Essential: true,
+        Image: {
+          "Fn::Join": [
+            "",
+            [
+              "123456789012.dkr.ecr.us-east-1.",
+              {
+                Ref: "AWS::URLSuffix"
+              },
+              "/fake-repo:latest"
+            ]
+          ]
+        },
+        MemoryReservation: 512,
+        Name: containerName,
+        PortMappings: [
+          {
+            ContainerPort: 8080,
+            Protocol: "tcp"
+          }
+        ]
+      }
+    ]
+  };
+}
+
+function vpcCidrTest(cidr: string) {
+  return {
+    CidrBlock: cidr
+  };
+}
+
+function stepfunctionsEndpoint() {
+  return {
+    ServiceName: {
+      "Fn::Join": [
+        "",
+        [
+          "com.amazonaws.",
+          {
+            Ref: "AWS::Region"
+          },
+          ".states"
+        ]
+      ]
+    }
+  };
+}
+
+function stateMachineTest() {
+  return {
+    RoleArn: {
+      "Fn::GetAtt": [
+        "testconstructStateMachineRoleA396E5D3",
+        "Arn"
+      ]
+    },
+    DefinitionString: "{\"StartAt\":\"StartState\",\"States\":{\"StartState\":{\"Type\":\"Pass\",\"End\":true}}}",
+    LoggingConfiguration: {
+      Destinations: [
+        {
+          CloudWatchLogsLogGroup: {
+            LogGroupArn: {
+              "Fn::GetAtt": [
+                "testconstructStateMachineLogGroup2EB4F48B",
+                "Arn"
+              ]
+            }
+          }
+        }
+      ],
+      Level: "ERROR"
+    }
+  };
+}
+
+function cloudwatchTest() {
+  return {
+    ComparisonOperator: "GreaterThanOrEqualToThreshold",
+    EvaluationPeriods: 1,
+    AlarmDescription: "Alarm for the number of executions that aborted exceeded the threshold of 1. ",
+    Dimensions: [
+      {
+        Name: "StateMachineArn",
+        Value: {
+          Ref: "testconstructStateMachine3333AAA9"
+        }
+      }
+    ],
+    MetricName: "ExecutionsAborted",
+    Namespace: "AWS/States",
+    Period: 300,
+    Statistic: "Maximum",
+    Threshold: 1
+  };
+}
+
+function checkForPublicPrivateVpc(stack: cdk.Stack) {
+  expect(stack).toHaveResourceLike('AWS::EC2::InternetGateway', {});
+  expect(stack).toCountResources('AWS::EC2::VPC', 1);
+  expect(stack).toCountResources('AWS::StepFunctions::StateMachine', 1);
+  expect(stack).toCountResources('AWS::ECS::Service', 1);
+}
+
+function checkForIsolatedVpc(stack: cdk.Stack) {
+  expect(stack).not.toHaveResourceLike('AWS::EC2::InternetGateway', {});
+  expect(stack).toCountResources('AWS::EC2::VPC', 1);
+  expect(stack).toCountResources('AWS::StepFunctions::StateMachine', 1);
+  expect(stack).toCountResources('AWS::ECS::Service', 1);
+}
+
+function checkForBaseResources(stack: cdk.Stack) {
```

Comment:
Not a fan of this - makes it difficult to see what each test is looking for. Let's not do this here, and going forward - our new style of checking 1 thing in each test should eliminate our need to have lots of checks duplicated between tests.

Reply:
If you'd rather change to the new testing style here rather than replacing all these statements, that's fine (although I think replacing the statements would be quicker, even if it annoys... :-) )